### PR TITLE
feat(ui): extract recorder sidebar into a window

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: utils/build/build.js [hunk 1]
```diff
@@ -17,6 +17,7 @@
 const child_process = require('child_process');
 const path = require('path');
 const chokidar = require('chokidar');
+const { platform } = require('os');
```
Comment (dgozman) [inline utils/build/build.js:20]: Unused?

File: utils/build/build.js [hunk 2]
```diff
@@ -122,4 +123,11 @@ onChanges.push({
   script: 'utils/generate_types/index.js',
 });
 
+// Copy images.
+steps.push({
```
Comment (dgozman) [inline utils/build/build.js:127]: You'd want to chokidar this in watch mode.

File: src/server/supplements/recorder/recorderApp.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    const icon = await readFileAsync(require.resolve('../../../../lib/web/recorder/app_icon.png'));
+    const crPopup = this._page._delegate as CRPage;
+    await crPopup._mainFrameSession._client.send('Browser.setDockTile', {
+      image: icon.toString('base64')
+    });
+
+    await this._page._setServerRequestInterceptor(async route => {
+      if (route.request().url().startsWith('https://playwright/')) {
+        const uri = route.request().url().substring('https://playwright/'.length);
+        const file = require.resolve('../../../../lib/web/recorder/' + uri);
```
Comment (dgozman) [inline src/server/supplements/recorder/recorderApp.ts:47]: Why not just './web/recorder/' ?

File: src/server/supplements/recorder/recorderApp.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  '.js': 'application/javascript',
+  '.png': 'image/png',
+  '.ttf': 'font/ttf',
+  '.svg': 'image/svg+xml',
+  '.webp': 'image/webp',
+  '.woff': 'font/woff',
+  '.woff2': 'font/woff2',
+};
+
+const DEFAULT_ARGS = [
```
Comment (dgozman) [inline src/server/supplements/recorder/recorderApp.ts:126]: Please import these from chromium.ts.

File: src/server/supplements/recorder/recorderApp.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  static async open(inspectedPage: Page): Promise<RecorderApp> {
+    const bounds = await CRPage.mainFrameSession(inspectedPage).windowBounds();
+    const recorderPlaywright = createPlaywright(true);
+    const context = await recorderPlaywright.chromium.launchPersistentContext('', {
+      ignoreAllDefaultArgs: true,
+      args: [
+        ...DEFAULT_ARGS,
+        `--user-data-dir=${path.join(os.homedir(),'.playwright-recorder')}`,
```
Comment (dgozman) [inline src/server/supplements/recorder/recorderApp.ts:81]: Why not use our Caches folder? We know it's a good one.

**Review by dgozman — APPROVED** [review]

File: src/server/browserContext.ts [hunk 6]
```diff
@@ -244,8 +244,6 @@ export abstract class BrowserContext extends EventEmitter {
 
   async _loadDefaultContext(progress: Progress) {
     const pages = await this._loadDefaultContextAsIs(progress);
-    if (pages.length !== 1 || pages[0].mainFrame().url() !== 'about:blank')
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply src/server/browserContext.ts:247]: _defaultArgs in chromium has the same check, it was a dupe.

File: utils/build/build.js (hunk 1 above)
Comment (pavelfeldman) [reply utils/build/build.js:20]: Done

File: utils/build/build.js (hunk 2 above)
Comment (pavelfeldman) [reply utils/build/build.js:127]: I don't anticipate this changing live

File: src/server/supplements/recorder/recorderApp.ts (hunk 3 above)
Comment (pavelfeldman) [reply src/server/supplements/recorder/recorderApp.ts:47]: But there is nothing in that folder?

File: src/server/supplements/recorder/recorderApp.ts (hunk 4 above)
Comment (pavelfeldman) [reply src/server/supplements/recorder/recorderApp.ts:126]: Done

File: src/server/supplements/recorder/recorderApp.ts (hunk 5 above)
Comment (pavelfeldman) [reply src/server/supplements/recorder/recorderApp.ts:81]: well, we definitely know that the home folder is good as well and conceptually this is user data, not program data.

**Pull request merged**